### PR TITLE
Provide iamin reference implementation

### PR DESCRIPTION
**Pull request by Nyrio**
iamin is an extension to blas and it is not included in `libopenblas.so` when cross-compiling for ARM. Therefore we need to provide our own reference implementation.

### CODE REVIEW

File: include/utils/system_reference_blas.hpp [hunk 1]
```diff
@@ -28,13 +28,27 @@
...
+#include <cmath>
 
-extern "C" {
-// i*amin is an extension
-CBLAS_INDEX cblas_isamin(const int N, const float *X, const int incX);
-CBLAS_INDEX cblas_idamin(const int N, const double *X, const int incX);
-CBLAS_INDEX cblas_icamin(const int N, const void *X, const int incX);
-CBLAS_INDEX cblas_izamin(const int N, const void *X, const int incX);
+// i*amin is an extension, provide an implementation
+template<typename scalar_t>
```
Comment (RossBrunton) [inline include/utils/system_reference_blas.hpp]: Can this be in an anonymous namespace if it's not public?

File: include/utils/system_reference_blas.hpp [hunk 2]
```diff
@@ -28,13 +28,27 @@
...
+inline int iamin(const int N, const scalar_t *X, const int incX)
+{
+  int best = 0;
+  for(int i = 1; i < N; i++) {
```
Comment (RossBrunton) [inline include/utils/system_reference_blas.hpp]: You aren't handling incX.

Comment (Nyrio) [reply include/utils/system_reference_blas.hpp]: Oh, my bad. Fixed

File: include/utils/system_reference_blas.hpp (hunk 1 above)
Comment (Nyrio) [reply include/utils/system_reference_blas.hpp]: Done

File: include/utils/system_reference_blas.hpp [hunk 3]
```diff
@@ -79,6 +72,27 @@ CBLAS_DIAG c_diag(char x) {
...
 }
+
+// i*amin is an extension, provide an implementation
+template<typename scalar_t>
+inline int iamin(const int N, const scalar_t *X, const int incX)
+{
+  int best = 0;
+  for(int i = 1; i < N; i += incX) {
+    if(std::abs(X[i]) < std::abs(X[best])) {
+      best = i;
```
Comment (RossBrunton) [inline include/utils/system_reference_blas.hpp]: This should return i / incX

Comment (Nyrio) [reply include/utils/system_reference_blas.hpp]: Thanks for catching that @RossBrunton , fixed

Comment (RossBrunton) [reply include/utils/system_reference_blas.hpp]: lgtm

**Comment by RossBrunton** [comment]
I see no isssues.

**Pull request merged**